jekyll/_cci2/browser-testing.md

```diff
@@ -0,0 +1,239 @@
+
+## Debugging Browser Tests
+
+Integration tests can be hard to debug, especially when they're running on a remote machine.
+This section provides some examples of how to debug browser tests on CircleCI.
+
+### Using Screenshots and Artifacts
+
+CircleCI may be configured to collect [build artifacts]( {{ site.baseurl }}/2.0/artifacts/)
+and make them available from your build. For example, artifacts enable you to save screenshots as part of your build,
+and view them when the build finishes. You must explicitly collect those files with the `store_artifacts` step and specify the `path` and `destination`, see the [store_artifacts]( {{ site.baseurl }}/2.0/configuration-reference/#store_artifacts) section of the Configuration Reference for an example.
+
+Saving screenshots is straightforward: it's a built-in feature in WebKit and Selenium, and is supported by most test suites:
+
+*   [Manually, using Selenium directly](http://docs.seleniumhq.org/docs/04_webdriver_advanced.jsp#remotewebdriver)
+*   [Automatically on failure, using Cucumber](https://github.com/mattheworiordan/capybara-screenshot)
+*   [Automatically on failure, using Behat and Mink](https://gist.github.com/michalochman/3175175)
+
+### Using a Local Browser to Access HTTP server on CircleCI
+
+If you are running a test that runs an HTTP server on CircleCI, sometimes it is helpful to use a browser running on your local machine to debug a
+failing test. Setting this up is easy with an SSH-enabled
+run.
+
+First, run an SSH build using the Rerun Job with SSH button on the Builds page of the CircleCI app. The command to log into
+the container over SSH apears, as follows:
+
+```
+ssh -p 64625 ubuntu@54.221.135.43
+```
+
+To add port-forwarding to the command, use the `-L` flag.
+The following example forwards
+requests to `http://localhost:3000` to port `8080` on the CircleCI container.
+This would be useful, for example, if your build runs a debug Ruby on Rails app, which listens
+on port 8080.
+
+```
+ssh -p 64625 ubuntu@54.221.135.43 -L 3000:localhost:8080
+```
+
+Then, open your browser on your local machine and navigate to
+`http://localhost:8080` to send requests directly to the server
+running on port `3000` on the CircleCI container. You can also manually start the
+test server on the CircleCI container (if it is not already running), and you
+should be able to access the running test server from the browser on your development machine.
+
+This is a very easy way to debug things when setting up Selenium tests, for
+example.
+
+### Interacting With the Browser Over VNC
+
+VNC allows you to view and interact with the browser that is running your tests. This will only work if you're using a driver that runs a real browser. You will be able to interact with a browser that Selenium controls, but phantomjs is headless &mdash; there is nothing to interact with.
+
+Before you start, make sure you have a VNC viewer installed. If you're using macOS, we recommend
```

Comment:
Do "we recommend"? Or should they "consider"? ;)

Reply:
yes, consider this!